internal/bminventory/inventory.go

```diff
@@ -458,6 +458,13 @@ func (b *bareMetalInventory) RegisterClusterInternal(
 		monitoredOperators = append(monitoredOperators, newOLMOperators...)
 	}
 
+	platform := params.NewClusterParams.Platform
+	if models.PlatformTypeBaremetal == *platform.Type && *params.NewClusterParams.UserManagedNetworking {
+		*platform.Type = models.PlatformTypeNone
+	} else if models.PlatformTypeNone == *platform.Type {
```

Comment:
Consider moving all this to a function than handles the platform and UMN params.
e.g. `alignClusterPlatformParams`